frontend/src/deviceDetailsPage/configuration/UploadConfigDialogOpenButton.jsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Button } from "@material-ui/core";
+import { CloudUpload } from "@material-ui/icons";
+
+import UploadConfigDialog from "./UploadConfigDialog";
+
+import DeviceInfo from "../../model/DeviceInfo";
+
+export default class UploadConfigDialogOpenButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.dialogElement = React.createRef();
+    this.openDialog = this.openDialog.bind(this);
+  }
+
+  openDialog() {
+    this.dialogElement.current.openDialog();
+  }
+
+  render() {
+    const {
+      device: { serialNumber }
+    } = this.props;
```

Comment:
does directly getting the `serialNumber` from the details page not work? it doesn't seem like you need anything beyond that for upload to work

Reply:
not sure if i'm going to need other information from the device since the backend isn't defined yet